lib/net.c

```diff
@@ -1200,11 +1201,80 @@ start_iface(struct gatekeeper_if *iface, unsigned int num_attempts_link_get)
 	return ret;
 }
 
+static void
+calculate_net_configure_para(struct net_config *net_conf)
+{
+	int i;
+
+	/*
+	 * The number of elements in the mbuf pool.
+	 *
+	 * The optimum size (in terms of memory usage) for a mempool is when
+	 * it is a power of two minus one.
+	 *
+	 * Need to provision enough memory for the worst case,
+	 * since each queue needs at least
+	 * gatekeeper_num_rx_desc + gatekeeper_num_tx_desc +
+	 * gatekeeper_max_pkt_burst descriptors. i.e.,
+	 * GATEKEEPER_DESC_PER_QUEUE = (gatekeeper_num_rx_desc +
+	 * 	gatekeeper_num_tx_desc + gatekeeper_max_pkt_burst).
```

Comment:
I think the l3fwd sample application provides a more detailed overview of what is needed here:

https://github.com/cjdoucette/dpdk/blob/46a945c1a3bf5904cb88c28bf1e8c441e0536641/examples/l3fwd/main.c#L481-L493

It's essentially the number of RX descriptors (across all queues for all ports), the number of TX descriptors (across all queues for all ports), the number of packet burst buffers (across all lcores for all bursts), the number of slots in the packet buffer cache (across all lcores).

Since it calculates it individually for the RX descriptors and TX descriptors, I think this would allow us to drop `gatekeeper_num_desc` and just use `gatekeeper_num_rx_desc` and `gatekeeper_num_tx_desc`.